### DISCUSSION TITLE
Is it possible to build the koreader on the RISC-V architecture?

### DISCUSSION THREAD

**Discussion by kyatsen** [koreader/koreader#11407] [Q&A] [answered]
I'd like to use Milk-V Duo SBC to interface with some SPI Eink Display.

However, this SBC is the RISC-V architecture.

The main application would be the koreader which I want to run on boot.

But, I'm not sure if it's can be cross-compiled for RISC-V

![image](https://github.com/koreader/koreader/assets/14968248/ccb8226a-cb89-4d1f-94ab-f76e976dfa7c)

**Comment by Frenzie** [comment]
I just posted a short answer over in https://github.com/koreader/koreader/issues/11406#issuecomment-1910919085

The longer answer is perhaps, with a lot of effort, if you rewrote large parts to use regular Lua.

**Reply by kyatsen** [reply]
Thanks @Frenzie for the answer.
Do you have perhaps recommendations for other ebook projects that cat be run on RISC-V?

Another question would be, is it possible to run koreader on raspberry pi zero?

**Reply by Frenzie** [reply, marked as answer]
I don't know much about the architecture other than that it exists.

For the second question I imagine one of the Debian arm packages should just work. (Try armhf first.)
